lib/Doctrine/DBAL/DBALException.php

```diff
@@ -28,21 +28,22 @@ class DBALException extends Exception
      */
     public static function driverExceptionDuringQuery(Driver $driver, Throwable $driverEx, $sql, array $params = [])
     {
-        $msg = "An exception occurred while executing '" . $sql . "'";
-        if ($params) {
-            $msg .= ' with params ' . self::formatParameters($params);
-        }
-        $msg .= ":\n\n" . $driverEx->getMessage();
+        $message = sprintf(
+            "An exception occurred while executing \"%s\"%s:\n\n%s.",
+            $sql,
+            $params !== [] ? sprintf(' with params %s', self::formatParameters($params)) : '',
+            $driverEx->getMessage()
+        );
 
-        return static::wrapException($driver, $driverEx, $msg);
+        return static::wrapException($driver, $driverEx, $message);
     }
 
     /**
      * @return self
      */
     public static function driverException(Driver $driver, Throwable $driverEx)
     {
-        return static::wrapException($driver, $driverEx, 'An exception occurred in driver: ' . $driverEx->getMessage());
+        return static::wrapException($driver, $driverEx, sprintf('An exception occurred in driver "%s".', $driverEx->getMessage()));
```

Comment:
I'm not sure we need to enclose sentences in quotes. When a message contains another message, the syntax with the colon makes more sense to me. The same syntax is used in Golang for error chaining.